pgjdbc/src/test/java/com/yugabyte/FallbackLBTestExtended.java

```diff
@@ -0,0 +1,214 @@
+    }
+  }
+
+  private static void checkNodeDownPrimary() throws SQLException {
+
+    executeCmd(path + "/bin/yb-ctl destroy", "Stop YugabyteDB cluster", 10);
+
+    executeCmd(path + "/bin/yb-ctl --rf 3 start --placement_info \"aws.us-west.us-west-1a\" ",
+        "Start YugabyteDB rf=3 cluster", 15);
+
+    String url = "jdbc:yugabytedb://127.0.0.1:5433,127.0.0.4:5433,127.0.0.7:5433/yugabyte?load-balance=true&topology-keys=";
+
+    try {
+      createConnectionsWithoutCloseAndVerify(url, "aws.us-west.*:1", expectedInput(6, 6, 6));
+
+      executeCmd(path + "/bin/yb-ctl stop_node 1", "Stop node 1", 10);
+      createConnectionsWithoutCloseAndVerify(url, "aws.us-west.*:1", expectedInput(-1, 15, 15));
+
+      executeCmd(path + "/bin/yb-ctl start_node 1 --placement_info \"aws.us-west.us-west-1a\"", "Start node 1", 10);
+      ClusterAwareLoadBalancer.forceRefresh = true;
+      try {
+        Thread.sleep(5000);
+      } catch (InterruptedException ie) {}
+      createConnectionsWithoutCloseAndVerify(url, "aws.us-west.*:1", expectedInput(16, 16, 16));
+
+    } finally {
+      executeCmd(path + "/bin/yb-ctl destroy", "Stop YugabyteDB cluster", 10);
+    }
+  }
+
+  private static void startYBDBClusterWithNineNodes() {
+    executeCmd(path + "/bin/yb-ctl destroy", "Stop YugabyteDB cluster", 10);
+
+    executeCmd(path + "/bin/yb-ctl --rf 3 start --placement_info \"aws.us-west.us-west-1a\" ",
+        "Start YugabyteDB rf=3 cluster", 15);
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.us-east.us-east-2a\"",
+        "Add a node", 10);
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.us-east.us-east-2a\"",
+        "Add a node", 10);
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.eu-north.eu-north-2a\"",
+        "Add a node", 10);
+
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.eu-west.eu-west-2a\"",
+        "Add a node", 10);
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.eu-west.eu-west-2a\"",
+        "Add a node", 10);
+    executeCmd(path + "/bin/yb-ctl add_node --placement_info \"aws.eu-north.eu-north-2a\"",
+        "Add a node", 10);
+
+    try {
+      Thread.sleep(5000);
+    } catch (InterruptedException ie) {}
+  }
+
+  private static void executeCmd(String cmd, String msg, int timeout) {
+    try {
+      ProcessBuilder builder = new ProcessBuilder();
+      builder.command("sh", "-c", cmd);
+      Process process = builder.start();
+      process.waitFor(timeout, TimeUnit.SECONDS);
+      int exitCode = process.exitValue();
+      if (exitCode != 0) {
+        throw new RuntimeException(msg + ": FAILED");
+      }
+      System.out.println(msg + ": SUCCEEDED!");
+    } catch (Exception e) {
+      System.out.println("Exception " + e);
+    }
+  }
+
+  private static void createConnectionsWithoutCloseAndVerify(String url, String tkValue, ArrayList<Integer> counts) throws SQLException {
```

Comment:
You can use the expected counts as a map instead of list, similar to the examples code.